src/Stratis.Bitcoin/Connection/ConnectionManager.cs

```diff
@@ -257,11 +258,69 @@ private void AddComponentStats(StringBuilder builder)
                     + " agent:" + agent);
             }
 
+            var oneTryBuilder = new StringBuilder();
+            var whiteListedBuilder = new StringBuilder();
+            var addNodeBuilder = new StringBuilder();
+            var otherBuilder = new StringBuilder();
+            var addNodeDict = this.ConnectionSettings.AddNode.ToDictionary(ep => ep.MapToIpv6(), ep => ep);
+
+            foreach (INetworkPeer peer in this.ConnectedPeers)
+            {
+                bool added = false;
+
+                var connectionManagerBehavior = peer.Behavior<ConnectionManagerBehavior>();
+                if (connectionManagerBehavior.OneTry)
+                {
+                    AddPeerInfo(oneTryBuilder, peer);
+                    added = true;
+                }
+
+                if (connectionManagerBehavior.Whitelisted)
+                {
+                    AddPeerInfo(whiteListedBuilder, peer);
+                    added = true;
+                }
+
+                if (addNodeDict.ContainsKey(peer.PeerEndPoint))
+                {
+                    AddPeerInfo(addNodeBuilder, peer);
+                    added = true;
+                }
+
+                if (!added)
+                {
+                    AddPeerInfo(otherBuilder, peer);
+                }
+            }
+
             int inbound = this.ConnectedPeers.Count(x => x.Inbound);
 
             builder.AppendLine();
             builder.AppendLine($"======Connection====== agent {this.Parameters.UserAgent} [in:{inbound} out:{this.ConnectedPeers.Count() - inbound}] [recv: {totalRead.BytesToMegaBytes()} MB sent: {totalWritten.BytesToMegaBytes()} MB]");
-            builder.AppendLine(peerBuilder.ToString());
+
+            if (oneTryBuilder.Length > 0)
+            {
+                builder.AppendLine(">>> Connect:");
```

Comment:
This is not correct onetry and connect are different things

Reply:
its called be called `onetry` ( we probably not need to specify `connect` when that is specified all other options are off (apart for whitelist).

What about cases where a peer is both `addnode` and `whitlisted`?

Reply:
Fixed.